Observer/OfflinePaymentObserver.php

```diff
@@ -39,32 +59,42 @@ public function execute(Observer $observer)
     {
         $order   = $observer->getEvent()->getOrder();
         $payment = $order->getPayment();
-
-        if ($payment->getAdditionalData('payment_type') !== 'bill_payment_tesco_lotus') {
-            return $this;
+        $emailData     = new \Magento\Framework\DataObject();
+        switch($payment->getAdditionalInformation('payment_type')) {
+            case 'bill_payment_tesco_lotus':
+                // make sure timezone is Thailand.
+                date_default_timezone_set("Asia/Bangkok");
+                $barcodeHtml   = $this->_helper->convertTescoSVGCodeToHTML($payment->getAdditionalInformation('barcode'));
+                $emailTemplate = 'send_email_tesco_template';
+                break;
+            case 'paynow':
+                $charge_id = $payment->getAdditionalInformation('charge_id');
+                $charge = $this->charge->find($charge_id);
+                date_default_timezone_set("Asia/Singapore");
+                $barcodeHtml   = "<img src= '".$charge->source['scannable_code']['image']['download_uri']."'/>";
+                $emailTemplate = 'send_email_paynow_template';
+                $emailData->setData(['banksUrl' => $this->_assetRepo->getUrl('Omise_Payment::images/supportedbanks.png')]);
+                break;
+            default:
+                return $this;
         }
-
         $paymentData   = $payment->getData();
 
         $amount        = number_format($paymentData['amount_ordered'], 2) . ' ' . $order->getOrderCurrency()->getCurrencyCode();
-        $barcodeHtml   = $this->_helper->convertTescoSVGCodeToHTML($payment->getAdditionalData('barcode'));
         $storeName     = $this->_scopeConfig->getValue('trans_email/ident_sales/name', \Magento\Store\Model\ScopeInterface::SCOPE_STORE);
         $storeEmail    = $this->_scopeConfig->getValue('trans_email/ident_sales/email', \Magento\Store\Model\ScopeInterface::SCOPE_STORE);
         $customerEmail = $order->getCustomerEmail();
         $orderId       = $order->getIncrementId();
 
-        // make sure timezone is Thailand.
-        date_default_timezone_set("Asia/Bangkok");
-
         // get DateTime deadline that is in next 24 hours.
-        $validUntil    = date("d-m-Y H:i:s" , time() + 24 * 60 * 60) . ' ICT';
+        $validUntil    = date("d-m-Y H:i:s" , time() + 24 * 60 * 60);
```

Comment:
This is not the deadline for PayNow charges and we can't assume 24 hours for all offline payment methods.  Instead, we should use the `expires_at` parameter of the charge.

Reply:
Done.